src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1387,30 +1384,25 @@ AddUnionAllSetOperationsToAttributeEquivalenceClass(AttributeEquivalenceClass **
 }
 
 
+#if PG_VERSION_NUM >= PG_VERSION_13
 /*
- * RangeTableOffsetCompat returns the range table offset(in glob->finalrtable) for the appendRelInfo.
- * For PG < 13 this is a no op.
+ * FindPreviousParentAmount returns the number of parents that come before
+ * the parent of the given append rel info.
  */
-static int
-RangeTableOffsetCompat(PlannerInfo *root, AppendRelInfo *appendRelInfo)
-{
-	#if PG_VERSION_NUM >= PG_VERSION_13
-	int i = 1;
-	for (; i < root->simple_rel_array_size; i++)
-	{
-		RangeTblEntry *rte = root->simple_rte_array[i];
-		if (rte->inh)
-		{
-			break;
+static int FindPreviousParentAmount(List* appendRelList, AppendRelInfo* targetAppendRelInfo) {
+	int targetParentIndex = targetAppendRelInfo->parent_relid;
+	Bitmapset *parent_ids = NULL;
+	AppendRelInfo *appendRelInfo = NULL;
+	foreach_ptr(appendRelInfo, appendRelList) {
+		int curParentIndex = appendRelInfo->parent_relid;
+		if (curParentIndex < targetParentIndex &&
+			!bms_is_member(curParentIndex, parent_ids)) {
+				parent_ids = bms_add_member(parent_ids, curParentIndex);
```

Comment:
minor: you don't need to check `bms_is_member `

Reply:
what do you mean?  You mean we don't need `!bms_is_member(curParentIndex, parent_ids)` ?

Reply:
yes, because bms should be fine even if you add the same member multiple times